src/lib/tabs/tab-body.ts

```diff
@@ -132,46 +143,45 @@ export class MatTabBody implements OnInit {
   /** The tab body content to display. */
   @Input('content') _content: TemplatePortal;
 
+  /** Position that will be used when the tab is immediately becoming visible after creation. */
+  @Input() origin: number;
+
   /** The shifted index position of the tab body, where zero represents the active center tab. */
   @Input()
   set position(position: number) {
-    if (position < 0) {
-      this._position = this._getLayoutDirection() == 'ltr' ? 'left' : 'right';
-    } else if (position > 0) {
-      this._position = this._getLayoutDirection() == 'ltr' ? 'right' : 'left';
-    } else {
-      this._position = 'center';
-    }
+    this._positionIndex = position;
+    this._computePositionAnimationState();
   }
-  _position: MatTabBodyPositionState;
-
-  /** The origin position from which this tab should appear when it is centered into view. */
-  @Input()
-  set origin(origin: number) {
-    if (origin == null) { return; }
 
-    const dir = this._getLayoutDirection();
-    if ((dir == 'ltr' && origin <= 0) || (dir == 'rtl' && origin > 0)) {
-      this._origin = 'left';
-    } else {
-      this._origin = 'right';
+  constructor(private _elementRef: ElementRef,
+              @Optional() private _dir: Directionality,
+              /**
+               * @deletion-target 7.0.0 changeDetectorRef to be made required.
+               */
+              changeDetectorRef?: ChangeDetectorRef) {
+
+    if (this._dir && changeDetectorRef) {
+      this._dir.change.subscribe(dir => {
```

Comment:
Shouldn't this assign to the `_dirChangeSubscription`? Also if you're doing this in the constructor, you don't have to initialize it to `Subscription.EMPTY`.

Reply:
Good catch. Missed that. Since `dir` is not always present, we need to either truthy check for the subscription or just use `Subscription.EMPTY`.

Reply:
Done.